xbmc/cores/AudioEngine/Sinks/pipewire/PipewireStream.cpp

```diff
@@ -85,17 +85,17 @@ bool CPipewireStream::TriggerProcess() const
   return true;
 }
 
-void CPipewireStream::Flush(bool drain)
+void CPipewireStream::Flush(bool drain) const
 {
   pw_stream_flush(m_stream.get(), drain);
 }
```

Comment:
A key that this function shouldn't be const is that it immediately calls a C function. C functions can't be marked const, so you shouldn't pretend to the outside world that they are.

Besides, flushing is a mutating operation, so const doesn't really apply here.

Reply:
I wouldn't be so strict regarding C functions. E.g. `CPipewireStream::GetState` calls `pw_stream_get_state` ([imp](https://github.com/PipeWire/pipewire/blob/2d5e0ef47ba51a15aef062afb3a2333f8ba90c3d/src/pipewire/stream.c#L1666-L1672)) which is absolutely fine to be used in a const method. (I would have preferred the `struct pw_stream*` parameter to be const but C devs often don't seem to care...)

Reply:
Fair.

> but C devs often don't seem to care

I should have finished with this disclaimer. Dependencies are usually found to be less const-correct, so it's not a reliable key. Indeed, what's important is the context, and getting state would be expected const, and flushing would be  expected non-const.